### PR TITLE
Added support for foreign keys.

### PR DESCRIPTION
**Pull request by flomar**
This patch adds support for foreign keys, including x11 lib (Arch Linux) stripped of debug symbols.

Foreign key support is disabled by default and enabled as follows:

```
var db = SQLITE.new()
db.foreign_keys = true
db.open_db()
```

Usage is pretty straightforward:

```
db.create_table("A", {
  "id": {"data_type": "int", "primary_key": true, "not_null": true},
  "name": {"data_type": "text", "not_null": true}
}

db.create_table("B", {
  "id": {"data_type": "int", "primary_key": true, "not_null": true},
  "a_id": {"data_type": "int", "foreign_key": "A.id", "not_null": true}
}
```
Please confirm it works for you, I don't want to be responsible for bugs. :)

It's working fine for me, constraints are applied by SQLITE as intended.

### CODE REVIEW

**Comment by 2shady4u** [comment]
Hello @flomar !

I'll be sure to test this on windows! 
If possible could you add an example that uses foreign keys inside of the "database.gd"
so these can be tested alongside all other SQLite features?

**Comment by flomar** [comment]
I added an example function to the "database.gd" file.

Let me know if you want something added.

File: demo/database.gd [hunk 1]
```diff
@@ -123,7 +123,51 @@ func _ready():
...
+
+# This example demonstrates the in-memory and foreign key support. It's 
+# rather contrived, but it gets the point across.
+func example_of_in_memory_and_foreign_key_support():
+    
+    # Create the database as usual.
+    db = SQLite.new()
+    # Enable in-memory storage.
+    db.path = ":memory:"
+    # Enable foreign keys.
```
Comment (2shady4u) [inline demo/database.gd]: Would recommend to add "db.verbose_mode = true" so the console receives actual information regarding the success/failure.

File: demo/database.gd [hunk 2]
```diff
@@ -123,7 +123,51 @@ func _ready():
 	
 	# Close the imported database
 	db.close_db()
```
Comment (2shady4u) [inline demo/database.gd]: After this, add a call to "example_of_in_memory_and_foreign_key_support()" so the new functionalities get called automatically when running the scene :)

**Comment by flomar** [comment]
Added the changes you requested.

**Pull request merged**